### PR TITLE
[FLINK-25537] [JUnit5 Migration] Module: flink-core with,Package: api

### PR DESCRIPTION
**Pull request by GOODBOY008**
Because flink-core is huge ， so split into multi pr to review.

Changes:

- Migrate mudule `flink-core`'s `api` package to junit5.

### CODE REVIEW

**Comment by GOODBOY008** [comment]
@Jiabao-Sun PTAL

**Comment by GOODBOY008** [comment]
I have compared https://github.com/apache/flink/actions/runs/8395180921/job/22994258945 (flink master) and https://github.com/GOODBOY008/flink/actions/runs/8517183618/job/23327527159 (my repo). The unit test count is same.

**Review by Jiabao-Sun — CHANGES_REQUESTED** [review]
Thanks @GOODBOY008 for this contribution.
I left some comments, and I will need another round to complete the review.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/AbstractGenericTypeComparatorTest.java [hunk 1]
```diff
@@ -22,11 +22,12 @@
 import org.apache.flink.api.common.typeutils.TypeComparator;
 import org.apache.flink.api.common.typeutils.TypeSerializer;
 
-import org.junit.Ignore;
-import org.junit.Test;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/AbstractGenericTypeComparatorTest.java:26]: The `@Test` methods visibility can be adjusted to package default.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/AbstractGenericTypeSerializerTest.java [hunk 2]
```diff
@@ -22,7 +22,7 @@
 import org.apache.flink.api.common.typeutils.TypeSerializer;
 import org.apache.flink.util.StringUtils;
 
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/AbstractGenericTypeSerializerTest.java:25]: The `@Test` methods visibility can be adjusted to package default.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerSnapshotTest.java [hunk 3]
```diff
@@ -146,47 +145,46 @@ public void testRestoreSerializerWithRemovedFields() {
                                 mockRemovedField(HEIGHT_FIELD)));
 
         final TypeSerializer<TestPojo> restoredSerializer = testSnapshot.restoreSerializer();
-        assertTrue(restoredSerializer.getClass() == PojoSerializer.class);
+        assertThat(restoredSerializer.getClass() == PojoSerializer.class).isTrue();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerSnapshotTest.java:148]: instanceOf is better.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerSnapshotTest.java [hunk 4]
```diff
@@ -146,47 +145,46 @@ public void testRestoreSerializerWithRemovedFields() {
...
 
     @Test
-    public void testRestoreSerializerWithNewFields() {
+    void testRestoreSerializerWithNewFields() {
         final PojoSerializerSnapshot<TestPojo> testSnapshot =
                 buildTestSnapshot(Collections.singletonList(HEIGHT_FIELD));
 
         final TypeSerializer<TestPojo> restoredSerializer = testSnapshot.restoreSerializer();
-        assertTrue(restoredSerializer.getClass() == PojoSerializer.class);
+        assertThat(restoredSerializer.getClass() == PojoSerializer.class).isTrue();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerSnapshotTest.java:170]: instanceOf is better.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerTest.java [hunk 5]
```diff
@@ -173,11 +174,8 @@ public boolean equals(Object other) {
                 return false;
             }
 
-            if ((nestedClass == null && otherTUC.nestedClass != null)
-                    || (nestedClass != null && !nestedClass.equals(otherTUC.nestedClass))) {
-                return false;
-            }
-            return true;
+            return (nestedClass != null || otherTUC.nestedClass == null)
+                    && (nestedClass == null || nestedClass.equals(otherTUC.nestedClass));
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerTest.java:178]: IMO, these changes are not necessary and unrelated to the JUnit migration.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerUpgradeTestSpecifications.java [hunk 6]
```diff
@@ -229,19 +228,23 @@ public TypeSerializer<StaticSchemaPojo> createUpgradedSerializer() {
...
 
         @Override
-        public Matcher<StaticSchemaPojo> testDataMatcher() {
-            return is(new StaticSchemaPojo("Gordon", 27, StaticSchemaPojo.Color.BLUE, false));
+        public Condition<StaticSchemaPojo> testDataCondition() {
+            return new Condition<>(
+                    value ->
+                            new StaticSchemaPojo("Gordon", 27, StaticSchemaPojo.Color.BLUE, false)
+                                    .equals(value),
+                    "");
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerUpgradeTestSpecifications.java:241]: Can we provide a meaningful description?
There are many similar cases in this class.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSubclassSerializerTest.java [hunk 7]
```diff
@@ -98,10 +98,7 @@ public boolean equals(Object other) {
             if (dumm1 != otherTUC.dumm1) {
                 return false;
             }
-            if (!dumm2.equals(otherTUC.dumm2)) {
-                return false;
-            }
-            return true;
+            return dumm2.equals(otherTUC.dumm2);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSubclassSerializerTest.java:101]: We don't need these changes in this PR.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/RowComparatorTest.java [hunk 8]
```diff
@@ -159,7 +160,7 @@ public boolean equals(Object o) {
 
             MyPojo myPojo = (MyPojo) o;
 
-            return name != null ? name.equals(myPojo.name) : myPojo.name == null;
+            return Objects.equals(name, myPojo.name);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/RowComparatorTest.java:163]: We don't need these changes int this PR.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/SubclassFromInterfaceSerializerTest.java [hunk 9]
```diff
@@ -103,10 +103,7 @@ public boolean equals(Object other) {
             if (dumm1 != otherTUC.dumm1) {
                 return false;
             }
-            if (!dumm2.equals(otherTUC.dumm2)) {
-                return false;
-            }
-            return true;
+            return dumm2.equals(otherTUC.dumm2);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/SubclassFromInterfaceSerializerTest.java:106]: We don't need these changes in this PR.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/kryo/KryoSerializerSnapshotTest.java [hunk 10]
```diff
@@ -32,32 +32,32 @@
...
-    @Before
-    public void setup() {
+    @BeforeEach
+    void setup() {
         oldConfig = new SerializerConfigImpl();
         newConfig = new SerializerConfigImpl();
     }
 
     @Test
     public void sanityTest() {
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/kryo/KryoSerializerSnapshotTest.java:59]:
```suggestion
   void sanityTest() {
```

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/tuple/base/TuplePairComparatorTestBase.java [hunk 11]
```diff
@@ -21,40 +21,31 @@
...
-public abstract class TuplePairComparatorTestBase<T extends Tuple, R extends Tuple>
-        extends TestLogger {
+public abstract class TuplePairComparatorTestBase<T extends Tuple, R extends Tuple> {
 
     protected abstract TypePairComparator<T, R> createComparator(boolean ascending);
 
     protected abstract Tuple2<T[], R[]> getSortedTestData();
 
     @Test
     public void testEqualityWithReference() {
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/tuple/base/TuplePairComparatorTestBase.java:42]:
```suggestion
   void testEqualityWithReference() {
```

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @GOODBOY008 for the quick fix. 
LGTM now.

**Comment by Jiabao-Sun** [comment]
@flinkbot run azure

**Pull request merged**